app/src/main/java/app/revanced/integrations/youtube/patches/components/KeywordContentFilter.java

```diff
@@ -0,0 +1,156 @@
+package app.revanced.integrations.youtube.patches.components;
+
+import static app.revanced.integrations.shared.StringRef.str;
+import static app.revanced.integrations.youtube.ByteTrieSearch.convertStringsToBytes;
+
+import android.os.Build;
+
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import app.revanced.integrations.shared.Logger;
+import app.revanced.integrations.shared.Utils;
+import app.revanced.integrations.youtube.ByteTrieSearch;
+import app.revanced.integrations.youtube.settings.Settings;
+
+/**
+ * <pre>
+ * Allows hiding home feed and search results based on keywords and/or channel names.
+ *
+ * Limitations:
+ * - Searching for a keyword phrase will give no search results
+ * - Filtering a channel name can still show Shorts from that channel in the search results
+ * - Some layout component residue will remain, such as the video chapter previews for some search results.
```

Comment:
While it is clear that limitations exist, it is not clear why they exist. Adding an explanation for the limitations would solve this.